File: docs-examples/pubsub-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/pubsub/BookCreatedListener.kt
package io.micronaut.configuration.lettuce.docs.pubsub

import io.micronaut.configuration.lettuce.pubsub.annotation.MessageChannel
import io.micronaut.configuration.lettuce.pubsub.annotation.RedisListener
import io.micronaut.messaging.annotation.MessageBody

@RedisListener
class BookCreatedListener {

    // tag::listener[]
    @MessageChannel("books.created")
    fun receive(@MessageBody event: BookCreated, @MessageChannel channel: String) {
        println("Received ${event.title} from $channel")
    }
    // end::listener[]
}

File: docs-examples/pubsub-python/src/test/python/micronaut/configuration/lettuce/docs/pubsub/BookCreatedListener.py
from typing import Annotated

from micronaut.configuration.lettuce.pubsub.annotation import MessageChannel, RedisListener
from micronaut.messaging.annotation import MessageBody

from .BookCreated import BookCreated


@RedisListener
class BookCreatedListener:

    # tag::listener[]
    @MessageChannel("books.created")
    def receive(self, event: Annotated[BookCreated, MessageBody], channel: Annotated[str, MessageChannel]) -> None:
        print(f"Received {event.title} from {channel}")
    # end::listener[]
